### DISCUSSION TITLE
Custom Search Options in Select

### DISCUSSION THREAD

**Discussion by kombiahrk** [filamentphp/filament#9279] [General]
for example 
i have product table with id, name, barcode and some extra fields
now i am fetching 
Select::make('product_id')
          ->label('Products')
          ->options(Product::query()->pluck('name', 'id'))
          ->searchable()

Now i get the name as result and search the name also i get the result but i want search using the barcode also.
so kindly guide me

**Comment by sinnbeck** [comment]
https://filamentphp.com/docs/3.x/forms/fields/select#returning-custom-search-results 

There is a method named `getSearchResultsUsing()`

**Reply by kombiahrk** [reply]
thanks i achived the result which i want but data retrieving is slow

**Reply by sinnbeck** [reply]
Ok. Maybe you have some bad indexes on you table or similar. Hard to guess

**Comment by markosole** [comment]
Hi all, I am on very similar road to @kombiahrk having products search as well as himself. Exact same thing is that it is slow. There is a way if you don't have many products, you can add `->preload()` which will then load all products. Not ideal if you have large data set. 

As I am on the similar problem, I have 4000 products and searching Barcode with Name as well. I am thinking of creating custom search form field that will query (more likely faster) and list found products as a <ul> list and then product can be picked with action button maybe. 
I am just thinking what would be best way to solve it as there's not pre-defined filament field that can do this.

Any thoughts on this?
